### DISCUSSION TITLE
MIDI Host

### DISCUSSION THREAD

**Discussion by volca02** [hathach/tinyusb#1084] [General]
Hi,

I'd like to start a project on rpi pico with it acting as a MIDI USB HOST device. Are there plans to introduce MIDI Host capabilities to this library?

Thanks

**Comment by hathach** [comment]
yes, my plan is **waiting** for more people to contribute to the project.

**Reply by volca02** [reply]
Fair enough, thanks for the reply. I may try working on it myself, even though the barrier to understand the project and concepts involved seems quite high.

**Reply by hathach** [reply]
You could always ask for help while doing the PR. Just submit it as draft PR, it will serve as code discussion

**Reply by csBlueChip** [reply]
I have a pull request for a WiP of a MIDI Host ...A MIDI "engine" to act as {synth [input], keyboard [output], router [throughput]} should handle *ALL* MIDI events (for which I could find documentation). It is written and compiles in a test harness - but is untested against devices, as I have yet to work out how to get the class/driver talking to my USB-MIDI adaptor.
